docs/source/contributing.md

```diff
@@ -120,6 +120,38 @@ git rebase -i HEAD~5
 
 Once your commit history is clean, ensure you have based on the [latest upstream](#updating-your-fork) before you open the PR.
 
+### Commit messages
+
+Please make the first line of your commit message a summary of the change that a user (not a developer) of Operator would like to read, 
```

Comment:
This is a bit too strict. Not all changes are user visible. For example, it is good practice to split changes into multiple patches, so we can end up with a pull request that looks like:

1: Update a dependency
2: Regenerate some files
3: Implement awesome feature

Only the 3rd one is user visible.

Reply:
I agree, if anyone think that splitting some changes into separate commit bending this rule is beneficial, I'm all in. I don't think this needs any mention in guide.